Proposal for a full tooling guide

emmanuelbernard opened the pull request:
See #1156

emmanuelbernard committed:
Issue 1156 - Rename ide-configuration guide into tooling guide

emmanuelbernard committed:
Issue 1156 - Rework the ide-configuration guide into a full tooling guide

gsmet reviewed (changes requested):
I spotted a couple of typos.

Looks good otherwise.

gsmet committed:
Fix a few typos

cescoffier reviewed (changes requested):
I've made a few comments. Probably old stuff that we missed.

gsmet committed:
Address Clement's comments

gsmet committed:
Re-add versions used in the documentation to the build-parent pom

Properties are not inherited from the BOM so unfortunately, we need to
define the versions twice.

As the documentation versions will be defined in the website, they
will be duplicated anyway so that's not an issue.

gsmet committed:
graal-version should be graalvm-version to be consistent with the README
